lib/membrane/remote_controlled/pipeline.ex

```diff
@@ -0,0 +1,426 @@
+  It is required to firstly use the `Membrane.RemoteControlled.Pipeline.subscribe/2` to subscribe for given message before awaiting
+  for that message.
+
+  Usage example:
+    1) awaiting for the pipeline's playback state to change into `:playing`:
+    ```
+    Pipeline.await_playback_state(pipeline, :playing)
+    ```
+  """
+  @spec await_playback_state(pid, Membrane.PlaybackState.t()) ::
+          Membrane.RemoteControlled.Message.t()
+  def await_playback_state(pipeline, playback_state) do
+    do_await(pipeline, PlaybackState, state: playback_state)
+  end
+
+  @doc """
+  Awaits for the first `Membrane.RemoteControlled.Message()` wrapping the `Membrane.RemoteControlled.Message.StartOfStream()` message
+  with no further constraints, sent by the process with `pipeline` pid.
+  It is required to firstly use the `Membrane.RemoteControlled.Pipeline.subscribe/2` to subscribe for given message before awaiting
+  for that message.
+
+  Usage example:
+    1) awaiting for the first `start_of_stream` occuring on any pad of any element in the pipeline:
+    ```
+    Pipeline.await_start_of_stream(pipeline)
+    ```
+  """
+  @spec await_start_of_stream(pid) :: Membrane.RemoteControlled.Message.t()
+  def await_start_of_stream(pipeline) do
+    do_await(pipeline, StartOfStream)
+  end
+
+  @doc """
+  Awaits for the first `Membrane.RemoteControlled.Message()` wrapping the `Membrane.RemoteControlled.Message.StartOfStream()` message
+  concerning the given `element`, sent by the process with `pipeline` pid.
+  It is required to firstly use the `Membrane.RemoteControlled.Pipeline.subscribe/2` to subscribe for given message before awaiting
+  for that message.
+
+  Usage example:
+    1) awaiting for the first `start_of_stream` occuring on any pad of the `:element_id` element in the pipeline:
+    ```
+    Pipeline.await_start_of_stream(pipeline, :element_id)
+    ```
+  """
+  @spec await_start_of_stream(pid(), Membrane.Element.name_t()) ::
+          Membrane.RemoteControlled.Message.t()
+  def await_start_of_stream(pipeline, element) do
+    do_await(pipeline, StartOfStream, element: element)
+  end
+
+  @doc """
+  Awaits for the first `Membrane.RemoteControlled.Message()` wrapping the `Membrane.RemoteControlled.Message.StartOfStream()` message
+  concerning the given `element` and the `pad`, sent by the process with `pipeline` pid.
+  It is required to firstly use the `Membrane.RemoteControlled.Pipeline.subscribe/2` to subscribe for given message before awaiting
+  for that message.
+
+  Usage example:
+    1) awaiting for the first `start_of_stream` occuring on the `:pad_id` pad of the `:element_id` element in the pipeline:
+    ```
+    Pipeline.await_start_of_stream(pipeline, :element_id, :pad_id)
+    ```
+  """
+  @spec await_start_of_stream(pid(), Membrane.Element.name_t(), Membrane.Pad.name_t()) ::
+          Membrane.RemoteControlled.Message.t()
+  def await_start_of_stream(pipeline, element, pad) do
```

Comment:
```suggestion
  def await_start_of_stream(pipeline, element, pad \\ Macro.var(:_, __MODULE__)) do
```

Wouldn't something like this cover the above case as well?

Reply:
So finally I have decided to stick to this implementation